sequencers/modeshape-sequencer-video/src/main/java/org/modeshape/sequencer/video/VideoMetadata.java

```diff
@@ -0,0 +1,200 @@
+                    DemuxerStream stream = container.getStream(i);
+
+                    Decoder coder = stream.getDecoder();
+
+                    if (coder.getCodecType() == Type.MEDIA_AUDIO) {
+                        streamMetadata.setStreamType("audio");
+                    } else if (coder.getCodecType() == Type.MEDIA_VIDEO) {
+                        streamMetadata.setStreamType("video");
+                    } else {
+                        streamMetadata.setStreamType("unknown");
+                    }
+
+                    if (coder.getCodec() != null) {
+                        streamMetadata.setCodec(coder.getCodec().getName());
+                    }
+                    streamMetadata.setSamplerate(nullValueIfInvalid(coder.getSampleRate()));
+                    streamMetadata.setChannels(nullValueIfInvalid(coder.getChannels()));
+                    streamMetadata.setWidth(nullValueIfInvalid(coder.getWidth()));
+                    streamMetadata.setHeight(nullValueIfInvalid(coder.getHeight()));
+                    if (coder.getTimeBase() != null) {
+                        streamMetadata.setFramerate(nullValueIfInvalid(1.0 / coder.getTimeBase().getDouble()));
+                    }
+                    streams.add(streamMetadata);
+                }
+            return true;
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        return false;
+    }
+
+    private Integer nullValueIfInvalid( int value ) {
+        if (value > 0) {
+            return value;
+        } else {
+            return null;
+        }
+    }
+
+    private Double nullValueIfInvalid( double value ) {
+        if (value > 0) {
+            return value;
+        } else {
+            return null;
+        }
+    }
+
+    public Double getDuration() {
+        return duration;
+    }
+
+    public Integer getBitrate() {
+        return bitrate;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public String getComment() {
+        return comment;
+    }
+
+    public String getEncoder() {
+        return encoder;
+    }
+
+    public List<StreamMetadata> getStreams() {
+        return streams;
+    }
+
+    private int read( byte[] a ) throws IOException {
+        if (in != null) {
+            return in.read(a);
+        }
+        din.readFully(a);
+        return a.length;
+    }
+
+
+    /**
+     * Set the input stream to the argument stream (or file). Note that {@link java.io.RandomAccessFile} implements
+     * {@link java.io.DataInput}.
+     *
+     * @param dataInput the input stream to read from
+     */
+    public void setInput( DataInput dataInput ) {
+        din = dataInput;
+        in = null;
+    }
+
+    /**
+     * Set the input stream to the argument stream (or file).
+     *
+     * @param inputStream the input stream to read from
+     */
+    public void setInput( InputStream inputStream ) {
```

Comment:
this should be converted to a constructor, since it seems to always be required.